[docs] Add recipe for save and manage filters from the panel (#19361)
mui/mui-x: docs/data/data-grid/filtering-recipes/filtering-recipes.md

{{"demo": "FilteringLocalStorage.js", "bg": "inline", "defaultCodeOpen": false}}

## Save and manage filters from the panel

Create a custom filter panel by wrapping the `GridFilterPanel` component and pass it to the `slots.filterPanel` prop.

> In the demo below, the custom component lets users save and manage filters that are persisted in the local storage. You can replace the local storage with a more scalable solution like a server-side database.

Reviewer: ```suggestion
In the demo below, the custom component allows users to save and manage filters, which are stored in local storage.
For a more scalable approach, you can replace local storage with a server-side database.
```

Reviewer: Nice, I didn't know about this. 👍
